components/builder-worker/src/runner/mod.rs

```diff
@@ -0,0 +1,350 @@
+
+    pub fn build(&self, logger: &mut Logger) -> Result<PackageArchive> {
+        let args =
+            vec![OsString::from("-s"),
+                 OsString::from(self.workspace.src()),
+                 OsString::from("-r"),
+                 OsString::from(self.workspace.studio()),
+                 OsString::from("-k"),
+                 OsString::from("core"),
+                 OsString::from("build"),
+                 OsString::from(Path::new(self.workspace.job.get_project().get_plan_path())
+                     .parent()
+                     .unwrap())];
+        let command = match PackageInstall::load(&STUDIO_PKG, None) {
+            Ok(package) => format!("{}/hab-studio", package.paths().unwrap()[0].display()),
+            Err(_) => panic!("should install core/hab-studio because it's missing?"),
+        };
+        debug!("building, cmd={:?}, args={:?}", command, args);
+        let mut child = Command::new(command)
+            .args(&args)
+            .env_clear()
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("failed to spawn child");
+        if let Some(ref mut stdout) = child.stdout {
+            for line in BufReader::new(stdout).lines() {
+                let mut l: String = line.unwrap();
+                let eol: &str = "\n";
+                l = l + eol;
+                logger.log(l.as_bytes());
+            }
+        }
+        let exit_status = child.wait().expect("failed to wait on child");
+        debug!("build complete, status={:?}", exit_status);
+        if exit_status.success() {
+            try!(fs::rename(self.workspace.src().join("results"), self.workspace.out()));
+            self.workspace.last_built()
+        } else {
+            Err(Error::BuildFailure(exit_status.code().unwrap_or(-1)))
+        }
+    }
+
+    pub fn job(&self) -> &proto::Job {
+        &self.workspace.job
+    }
+
+    fn complete(&mut self) -> () {
+        self.teardown().err().map(|e| error!("{}", e));
+        self.workspace.job.set_state(proto::JobState::Complete);
+    }
+
+    fn fail(&mut self) -> () {
+        self.teardown().err().map(|e| error!("{}", e));
+        self.workspace.job.set_state(proto::JobState::Failed);
+    }
+
+    fn post_process(&self, archive: &mut PackageArchive) -> bool {
+        // JW TODO: In the future we'll support multiple and configurable post processors, but for
+        // now let's just publish to the public depot
+        //
+        // Things to solve right now
+        // * Where do we get the token for authentication?
+        //      * Should the workers ask for a lease from the JobSrv?
+        let client =
+            depot_client::Client::new(&hab_core::url::default_depot_url(), PRODUCT, VERSION, None)
+                .unwrap();
+        if let Some(err) = client.x_put_package(archive, &self.auth_token).err() {
+            error!("post processing error, ERR={:?}", err);
+            return false;
```

Comment:
I concur, while it might be handy at first to leave failed jobs around for debugging/troubleshooting, it'd get out of hand quickly.